pipeline/transformer/transformer_mets/src/main/scala/uk/ac/wellcome/platform/transformer/mets/parsers/MetsXmlParser.scala

```diff
@@ -48,4 +50,96 @@ object MetsXmlParser {
           new Exception("Found multiple accessCondtions in METS XML"))
     }
   }
+
+  /** Here we use the first defined item in the physicalStructMap to lookup a
+    *  file ID the fileObjects mapping, and use the files location as the
+    *  thumbnail image.
+    */
+  private def thumbnailLocation(root: Elem): Option[String] =
+    physicalStructMap(root).headOption
+      .flatMap {
+        case (_, fileId) =>
+          fileObjects(root).get(fileId)
+      }
+      .map(_.stripPrefix("objects/"))
+
+  /** The METS XML contains locations of associated files, contained in a
+    *  mapping with the following format:
+    *
+    *  <mets:fileSec>
+    *    <mets:fileGrp USE="OBJECTS">
+    *      <mets:file ID="FILE_0001_OBJECTS" MIMETYPE="image/jp2">
+    *        <mets:FLocat LOCTYPE="URL" xlink:href="objects/b30246039_0001.jp2" />
+    *      </mets:file>
+    *      <mets:file ID="FILE_0002_OBJECTS" MIMETYPE="image/jp2">
+    *        <mets:FLocat LOCTYPE="URL" xlink:href="objects/b30246039_0002.jp2" />
+    *      </mets:file>
+    *    </mets:fileGrp>
+    *  </mets:fileSec>
+    *
+    *  We need to look up the physicalStructMap to ascertain what each of these
+    *  files refer to.
+    */
+  private def fileObjects(root: Elem): Map[String, String] =
```

Comment:
So I think this would give me a mapping

```scala
Map("FILE_0001_OBJECTS" -> "objects/b30246039_0001.jp2", "FILE_0002_OBJECTS" -> "objects/b30246039_0002.jp2")
```

Reply:
Yes correct :+1: